docs/apis/plugintypes/tiny/testing.md

```diff
@@ -0,0 +1,181 @@
+
+:::note iFrames
+
+Behat makes use of WebDriver to control browsers. The WebDriver specification limits the scope of a command to the current _context_ where a context is a browser Window, Frame, or iFrame.
+
+TinyMCE places the Editor UI within its own iFrame. To control any of its interface, you _may_ need to switch to the iFrame for your editor first.
+
+Most of the steps written for TinyMCE are already aware of the iFrame.
+
+:::
+
+##### Working with toolbar buttons
+
+```gherkin title="Expanding all toolbar buttons"
+Given I expand all toolbars for the "Description" TinyMCE editor
+```
+
+:::caution
+
+The default TinyMCE configuration will automatically collapse the toolbar to show as many elements as will fit on the page.
+
+You should always use the "I expand all toolbars" step before interacting with a button on the toolbar as it may not be visible in all cases.
+
+This step is safe to call if the toolbar is already expanded, or does not need to be expanded.
+
+:::
+
+```gherkin title="Clicking on a button for your editor"
+When /^I click on the "Bold" button for the "Description" TinyMCE editor
+```
+
+```gherkin title="Checking the state of a button in your editor"
+Then the "Bold" button of the "Description" TinyMCE editor has state "true"
+```
+
+##### Working with the menu bar
+
+```gherkin title="Clicking on a menu bar button"
+I click on the "Insert" menu item for the "Description" TinyMCE editor
+```
+
+```gherkin title="Clicking on a nested menu bar button"
+I click on the "Insert > Image" menu item for the "Description" TinyMCE editor
+```
+
+##### Selecting content
+
+```gherkin title="Selecting content by tag name and index"
+I select the "p" element in position "2" of the "Description" TinyMCE editor
+```
+
+```gherkin title="Selecting content using a selector and type"
+I select the ".example_placeholder" "css_element" in the "Description" TinyMCE editor
+```
+
+:::note
+
+All content selection is performed within the context of the editor's own iframe. Please note that some of the standard Behat locators will not work as they expect certain elements to be present within the page body.
+
+:::
+
+### Writing your own steps
+
+If you are writing your own steps, then we recommend that you:
+
+- include the name of your plugin, or "TinyMCE editor" in the wording of the step so that it is not mis-used for another purpose
+- check that the Feature, or Scenario, includes the frankenstyle component name for your plugin
+
+```php title="Checking for a tag on your Feature or Scenario"
+if (!$this->has_tag('tiny_myplugin')) {
+    throw new DriverException(
+        'This step must have the @tiny_myplugin tag on either the Feature, or the Scenario.'
+    );
+}
+```
+
+### Important notes
+
+Some of these steps will be improved in coming changes. See MDL-75926 and MDL-76853 for more information.
```

Comment:
It seems there is a bug with sentences with more than one MDL because if you look at the preview, the first one is linked properly but the second one (MDL-76853) is displayed in plain text (so it's probably unrelated to this issue, but you know better how these MDL are filtered, so I thought it was better to raise it here).

Reply:
Good catch - addressed in https://github.com/moodle/devdocs/pull/522